src/replicache.ts

```diff
@@ -564,7 +574,7 @@ export default class Replicache implements ReadTransaction {
    * If an error occurs in the `body` the `onError` function is called if
    * present. Otherwise, the error is thrown.
    */
-  subscribe<R, E>(
+  subscribe<R extends ToJSONValue | undefined, E>(
```

Comment:
I think we can keep this as `R` (`R extends any`) but document that our equality check works like `JSON.stringify`. Seems mostly worth it to not break backwards compat.

Reply:
I'm trying to not change the API...

One approach would be to have `ScanResult` have `async every` that works like `Array.prototype.every` (stops when returning falsey) and maybe even `forEach`, and `some` for consistency) but keep the iterator.

I'm tempted to remove the `ReadableStreamDefaultController` since it is all sync anyway.

Reply:
I think it is worth breaking the API here to simplify.